numpydoc/validate.py

```diff
@@ -584,7 +584,15 @@ def validate(obj_name):
                     ("string", "str"),
                 ]
                 for wrong_type, right_type in common_type_errors:
-                    if wrong_type in doc.parameter_type(param):
```

Comment:
Can't we just change the `in` to `==` instead?

Reply:
Almost, but this needs to support more complex type descriptions than "str" such as "list of str". The regex could be improved to avoid the nested ifs. Let me know if you want me to give that a shot.

Reply:
Good point - then how about `if wrong_type in set(doc.parameter_type(param).split()):`? I tend toward trying to keep the logic as simple and short as possible, but if a regex is necessary then so be it!

Reply:
That was close and got me to where it needed to go. Your above doesn't handle punctuation (specifically `,`) so I changed it to use `re.split` to split on `\W`. That worked nicely.